src/Parser/Druid/Feral/Modules/Bleeds/RakeSnapshot.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import Snapshot, { JAGGED_WOUNDS_MODIFIER, PANDEMIC_FRACTION } from '../FeralCore/Snapshot';
+
+/*
+Identify inefficient refreshes of the Rake DoT:
+  Refresh of a Rake doing double damage due to Prowl with one that will not do double damage.
+  Early refresh of a Rake doing bonus damage from snapshot buffs with one that will do less damage.
+*/
+
+const RAKE_BASE_DURATION = 15000;
+class RakeSnapshot extends Snapshot {
+  rakeCastCount = 0;        // count of rake casts of all kinds
+  prowlLostCastCount = 0;   // count of rake buffed with Prowl ending early due to refresh without Prowl buff
+  prowlLostTimeSum = 0;     // total time cut out from the end of prowl-buffed rake bleeds by refreshing early (milliseconds)
+  downgradeCastCount = 0;   // count of rake DoTs refreshed with weaker snapshot before pandemic
+
+  hasBloodtalonsTalent = false;
+
+  on_initialized() {
+    this.spellCastId = SPELLS.RAKE.id;
+    this.durationOfFresh = RAKE_BASE_DURATION;
+    this.isProwlAffected = true;
+    this.isTigersFuryAffected = true;
+
+    const combatant = this.combatants.selected;
+    if (combatant.hasTalent(SPELLS.BLOODTALONS_TALENT.id)) {
+      this.isBloodtalonsAffected = true;
+      this.hasBloodtalonsTalent = true;
+    }
+    if (combatant.hasTalent(SPELLS.JAGGED_WOUNDS_TALENT.id)) {
+      this.durationOfFresh *= JAGGED_WOUNDS_MODIFIER;
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    if (SPELLS.RAKE.id === event.ability.guid) {
+      ++this.rakeCastCount;
+    }
+
+    super.on_byPlayer_cast(event);
+  }
+
+  checkRefreshRule(event, stateOld, stateNew) {
+    if (stateOld.prowl && !stateNew.prowl) {
+      // refresh removed a powerful prowl-buffed bleed
+      const timeLost = stateOld.expireTime - event.timestamp;
+      this.prowlLostTimeSum += timeLost;
+      ++this.prowlLostCastCount;
+
+      event.meta = event.meta || {};
+      event.meta.isInefficientCast = true;
+      event.meta.inefficientCastReason = `You lost ${(timeLost / 1000).toFixed(1)} seconds of a Rake empowered with Prowl by refreshing early.`;
+    } else if (stateOld.pandemicTime > event.timestamp &&
+      stateOld.power > stateNew.power &&
+      !stateNew.hasProwl) {
```

Comment:
if `stateOld.power > stateNew.power` isn't it still a bad refresh even though the new cast has Prowl? For instance if you do a rake with BT, TF and Prowl and then overwrite it right away with a rake only buffed by Prowl.

Reply:
Overwriting immediately like that would likely be a damage loss. But a downgrade while still maintaining Prowl is often an overall damage increase. The clearest case is when Incarnation (which gives the Prowl buff effect) is about to wear off.

Let's say the player already has a 2.6 (Prowl, BT, TF) Rake active, with`t`seconds remaining. Incarnation is about to wear off. Should they cast Rake at 2.0 (just Prowl)?

If they cast it, they'll have a 2.0 Rake for 15 seconds plus whatever pandemic they get from the existing, so `15 + min(4.5, t)`
If they don't cast it, they'll have a 2.6 Rake for`t`

We can model Rake for the next 20 second (long enough for the DoTs in question to expire), and assume when they wear off Rake will be immediately replaced by a 1.0 version. This interactive graph shows the scenario:
https://www.desmos.com/calculator/iovjnuz54k
The x axis is the`t`duration that was remaining when we refresh, and y is the total damage done over the modelled period.
A is the original powerful Rake. B is the new one we're considering casting. C is the non-prowl Rake that will fill in when they fade. 
The red line shows total Rake damage if we don't cast the new one, and the blue line shows if we do.

In this scenario it's a gain to refresh with a Rake weaker by both TF and BT so long as the existing one has less than 12 seconds remaining - and Rake's base duration is only 15!

With all that said, a player going from 2.6 to 2.0 on a long duration Rake when they have plenty of time left on Incarnation is probably a mistake. But it would take some more careful analysis to decide for certain.